src/oemof/solph/constraints/storage_level.py

```diff
@@ -0,0 +1,177 @@
+        OUTPUTS = po.Set(initialize=output_levels.keys())
+        setattr(model, f"{name}_OUTPUTS", OUTPUTS)
+
+        active_output = po.Var(
+            OUTPUTS, model.TIMESTEPS, domain=po.Binary, bounds=(0, 1)
+        )
+        setattr(model, f"{name}_active_output", active_output)
+
+        constraint_name = f"{name}_output_active_constraint"
+
+        def _output_active_rule(m):
+            r"""
+            .. math::
+                y_n \le E(t) / E_n
+            """
+            for t in m.TIMESTEPS:
+                for o in output_levels:
+                    getattr(m, constraint_name).add(
+                        (o, t),
+                        m.GenericStorageBlock.storage_content[
+                            storage_component, t + 1
+                        ]
+                        / storage_component.nominal_storage_capacity
+                        >= active_output[o, t] * output_levels[o],
+                    )
+
+        setattr(
+            model,
+            constraint_name,
+            po.Constraint(
+                OUTPUTS,
+                model.TIMESTEPS,
+                noruleinit=True,
+            ),
+        )
+        setattr(
+            model,
+            constraint_name + "build",
+            po.BuildAction(rule=_output_active_rule),
+        )
+
+        # Define constraints on the output flows
+        def _constraint_output_rule(m, o, p, t):
+            return (
+                m.flow[multiplexer_bus, o, p, t]
+                / m.flows[multiplexer_bus, o].nominal_value
+                <= active_output[o, t]
+            )
+
+        setattr(
+            model,
+            f"{name}_output_constraint",
+            po.Constraint(
+                OUTPUTS,
+                model.PERIODS,
+                model.TIMESTEPS,
+                rule=_constraint_output_rule,
+            ),
+        )
+
+    _outputs()
+
+    def _inputs():
+        INPUTS = po.Set(initialize=input_levels.keys())
+        setattr(model, f"{name}_INPUTS", INPUTS)
+
+        inactive_input = po.Var(
+            INPUTS, model.TIMESTEPS, domain=po.Binary, bounds=(0, 1)
+        )
+        setattr(model, f"{name}_active_input", inactive_input)
+
+        constraint_name = f"{name}_input_active_constraint"
+
+        def _input_active_rule(m):
+            r"""
+            .. math::
+                \hat{y}_n \ge (E(t) - E_n) - E_{max}
+            """
+            for t in m.TIMESTEPS:
+                for o in input_levels:
+                    getattr(m, constraint_name).add(
+                        (o, t),
+                        (
+                            m.GenericStorageBlock.storage_content[
+                                storage_component, t
+                            ]
+                            - input_levels[o]
+                        )
+                        / storage_component.nominal_storage_capacity
+                        <= inactive_input[o, t],
+                    )
+
```

Comment:
Formulation is different to the one in the docstring. This seems to need some fixing. Also, normalization looks somewhat odd.

Reply:
Actually, both formulations were wrong. I did an error copying the formula from the paper.